ComponentTextKitApplicationTests/CKTextKitTruncationTests.mm

```diff
@@ -140,4 +140,30 @@ - (void)testHandleZeroHeightConstrainedSize
                                                    constrainedSize:constrainedSize]);
 }
 
+- (void)testNoLimitOfMaximumNumberOfLines
+{
+  const NSUInteger maximumNumberOfLines = 0;
+  NSAttributedString *attributedString = [self _sentenceAttributedString];
+  CKTextKitRenderer *renderer = [[CKTextKitRenderer alloc]
+                                 initWithTextKitAttributes:{
+                                   .attributedString = attributedString,
+                                   .maximumNumberOfLines = maximumNumberOfLines,
+                                 }
+                                 constrainedSize:CGSizeMake(50, INFINITY)];
+  XCTAssertTrue(renderer.lineCount > maximumNumberOfLines);
+}
+
+- (void)testEnforcementOfMaximumNumberOfLines
+{
+  const NSUInteger maximumNumberOfLines = 3;
+  NSAttributedString *attributedString = [self _sentenceAttributedString];
+  CKTextKitRenderer *renderer = [[CKTextKitRenderer alloc]
+                                 initWithTextKitAttributes:{
+                                   .attributedString = attributedString,
+                                   .maximumNumberOfLines = maximumNumberOfLines,
+                                 }
+                                 constrainedSize:CGSizeMake(50, INFINITY)];
+  XCTAssertEqual(renderer.lineCount, maximumNumberOfLines);
+}
```

Comment:
Interesting, so this fixes the renderer reporting the wrong number of lines. How about rendering? Was our numberOfLines support there broken too?